### ISSUE TITLE
Not all "breaking changes" are truly breaking

### ISSUE THREAD

**Issue by hauleth** [getsentry/sentry-elixir#212]
Neither of these was "breaking change" in sense of semver:

- #201 
- #172 

Booth of these was backward compatible changes and [according to semver](http://semver.org/#spec-item-7) this should be minor version bump.

Such rapid major version bumping makes it hard to follow them in our projects and is quite inconvenient.

**Comment by hauleth** [comment]
Since version 4.0 there was only one truly breaking change and honestly I would do this in 2 step change. Firstly add new functionality and documentation to use new way for defining #190 and make deprecation warning for old option (minor version bump) and then after grace period remove option and bump major version.

**Comment by mitchellhenke** [comment]
@hauleth Thanks for opening this.  The feedback (and your past contributions) are very appreciated.

First, apologies for any inconvenience from the breaking changes.  #174 does unfortunately break backward compatibility they way I ended up doing it for those using a custom HTTP client, as the public `@behaviour` changed in arity.  I had some hesitation in bumping the major versions for `5.0.0` and `6.0.0`, but did so out of an abundance (of maybe too much) caution.

I can do a lot better to avoid breaking changes entirely, and plan to make more of an effort to use deprecations to maintain backward compatibility for longer periods of time.

Again, thanks for your contributions and feedback, and apologies for any inconvenience I've caused.
